apps/movies/src/movies.controller.ts

```diff
@@ -28,8 +31,25 @@ export class MoviesController {
   }
 
   @Get()
-  findAll() {
-    return this.moviesService.findAll();
+  async getPaginatedMovies(
+    @Query('page', new DefaultValuePipe(1), ParseIntPipe) page = 1,
```

Comment:
```suggestion
    @Query() queryParams: GetPaginatedMoviesQueryParamsDto,
```

So let's create a dto for query params:

import { IsNumber, IsOptional, Min, Max } from 'class-validator';
import { Transform } from 'class-transformer';

export class QueryCollateralTypeDto {
  @Transform(({ value }) => parseInt(value))
  @IsNumber()
  @Min(1)
  page: number = 1;

  @Transform(({ value }) => parseInt(value))
  @IsNumber()
  @Min(1)
  @Max(100)
  limit: number = 20;
}

So the idea is to create a separate dto for query params where you can define default value, add validation logic and transform logic